src/components/map/GoogleMap.svelte

```diff
@@ -51,20 +56,54 @@
 	async function loadStopsAndAddMarkers(lat, lng) {
 		const json = await loadStopsForLocation(lat, lng);
 		const stops = json.data.list;
+		allStops = [...allStops, ...stops];
 
-		for (const s of stops) {
-			if (markerExists(s)) {
-				continue;
-			}
+		clearAllMarkers();
 
-			addMarker(s);
+		if (showRoute && selectedRoute) {
+			const stopsToShow = allStops.filter((s) => s.routeIds.includes(selectedRoute.id));
+			stopsToShow.forEach((s) => addMarker(s));
+		} else {
+			stops.forEach((s) => addMarker(s));
 		}
 	}
 
-	function markerExists(s) {
-		return markers.some((marker) => marker.s.id === s.id);
+	function clearAllMarkers() {
+		markers.forEach(({ marker, overlay, element }) => {
+			if (marker) {
+				marker.setMap(null);
```

Comment:
this can be simplified to `marker?.setMap(null)`

Reply:
sure, done the optional chaining.